urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/ui/StaticPermFrame.java

```diff
@@ -0,0 +1,852 @@
+package permlab.ui;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.KeyboardFocusManager;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.util.ArrayList;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingConstants;
+import permlib.Permutation;
+import permlab.utilities.ExportInterface;
+import permlib.utilities.RestrictedPermutation;
+import permlib.utilities.IOUtilities;
+import permlab.utilities.PaintUtilities;
+
+/**
+ * Display single permutations (and their extensions)
+ *
+ * @author M Albert
+ */
+public class StaticPermFrame extends javax.swing.JFrame {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 49 other classes. By comparison, 98% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.